js/src/components/enhanced-conversion-tracking-settings/cta/index.test.js

```diff
@@ -0,0 +1,268 @@
+			isResolving: false,
+		} );
+
+		render( <CTA onAcceptTermsClick={ handleOnAcceptTerms } /> );
+
+		const button = screen.getByRole( 'button' );
+		userEvent.click( button );
+
+		expect( window.open ).toHaveBeenCalledTimes( 1 );
+		expect( handleOnAcceptTerms ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	test( 'Click on enable/confirm button callback', () => {
+		const handleOnEnable = jest.fn().mockName( 'On Enable click' );
+
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: true,
+			isResolving: false,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: null,
+			isResolving: false,
+		} );
+
+		render( <CTA onEnableClick={ handleOnEnable } /> );
+
+		const button = screen.getByRole( 'button' );
+		userEvent.click( button );
+
+		expect( handleOnEnable ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	test( 'Confirm/enable button callback should not be called if TOS has not been accepted', () => {
+		const handleOnEnable = jest.fn().mockName( 'On Enable click' );
+
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: false,
+			isResolving: false,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: ENHANCED_ADS_CONVERSION_STATUS.ENABLED,
+			isResolving: false,
+		} );
+
+		render( <CTA onEnableClick={ handleOnEnable } /> );
+
+		const button = screen.getByRole( 'button' );
+		userEvent.click( button );
+
+		expect( handleOnEnable ).not.toHaveBeenCalled();
+	} );
+
+	test( 'Click on disable button callback', () => {
+		const handleOnDisable = jest.fn().mockName( 'On Disable click' );
+
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: true,
+			isResolving: false,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: ENHANCED_ADS_CONVERSION_STATUS.ENABLED,
+			isResolving: false,
+		} );
+
+		render( <CTA onDisableClick={ handleOnDisable } /> );
+
+		const button = screen.getByRole( 'button' );
+		userEvent.click( button );
+
+		expect( handleOnDisable ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	test( 'Disable button callback should not be called if TOS has not been accepted', () => {
+		const handleOnDisable = jest.fn().mockName( 'On Disable click' );
+
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: false,
+			isResolving: false,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: ENHANCED_ADS_CONVERSION_STATUS.ENABLED,
+			isResolving: false,
+		} );
+
+		render( <CTA onDisableClick={ handleOnDisable } /> );
+
+		const button = screen.getByRole( 'button' );
+		userEvent.click( button );
+
+		expect( handleOnDisable ).not.toHaveBeenCalled();
+	} );
+
+	test( 'Should not the enable button if TOS has been accepted', () => {
```

Comment:
The description seems to be incorrect.

```suggestion
	test( 'Should render the enable button if TOS has been accepted and the status is not enabled', () => {
```

Reply:
👍

Reply:
👍